core/agents/listenbrainz/agent.go

```diff
@@ -108,6 +110,43 @@ func (l *listenBrainzAgent) IsAuthorized(ctx context.Context, userId string) boo
 	return err == nil && sk != ""
 }
 
+func (l *listenBrainzAgent) CanProxyStars(ctx context.Context, userId string) bool {
+	if !l.proxyStars {
+		return false
+	}
+	return l.IsAuthorized(ctx, userId)
+}
+
+func (l *listenBrainzAgent) CanStar(track *model.MediaFile) bool {
+	return track.MbzTrackID != ""
```

Comment:
The field got renamed to `MbzRecordingId` #2279 , so this needs to be adapted in this PR.

Another way to handle the situation of missing MBID would be to use the metadata lookup endpoint first. See:

https://listenbrainz.readthedocs.io/en/latest/users/api/metadata.html#get--1-metadata-lookup-
https://listenbrainz.readthedocs.io/en/latest/users/api-usage.html#lookup-mbids

As this using the same mapping that LB would also apply when submitting a listen without IDs I think it would be consistent.

Reply:
Personally, I'd rather not try and guess the correct metadata (that's basically what the endpoint does) for stars/loves, but I'll defer to Deluan on this.